index4_2.js

```diff
@@ -0,0 +1,42 @@
+/* с помощью map, raduce вывести url у которого большая площадь
+ */
+
+const boxarts = [
+  {
+    width: 200,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/imagin/2891/Fracture200.jpg",
+  },
+  {
+    width: 150,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/imagin/2891/Fracture150.jpg",
+  },
+  {
+    width: 300,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/imagin/2891/Fracture300.jpg",
+  },
+  {
+    width: 425,
+    height: 150,
+    url: "http://cdn-0.nflximg.com/imagin/2891/Fracture425.jpg",
+  },
+];
+
+const findMaxSquaret = (boxarts) => {
+  const arraySquares = boxarts.map(({ width, height, ...otherList }) => {
+    return {
+      square: width * height,
+      ...otherList,
+    };
+  });
+
+  return arraySquares.reduce((result, currentElement) => {
+    return result.square >= currentElement.square
+      ? result.url
+      : currentElement.url;
+  });
+};
```

Comment:
1. Your function works incorrect. try to put item with biggest square not in the end and you will se that your function return latest, not biggest
2. this task could be resolved via single reduce without map